qa/tasks/cephfs/test_mirroring.py

```diff
@@ -1157,6 +1157,10 @@ def exec_git_cmd(cmd_list):
         self.peer_add(self.primary_fs_name, self.primary_fs_id, "client.mirror_remote@ceph", self.secondary_fs_name)
 
         self.add_directory(self.primary_fs_name, self.primary_fs_id, f'/{repo_path}')
+
+        # clear git logs before taking snapshot
+        self.mount_a.run_shell(['rm', '-rf', f'{repo_path}/.git/logs'])
```

Comment:
This happens mainly during the incremental sync. The md5sum differs consistently for the files `.git/logs/HEAD` and `.git/logs/refs/heads/main` for the source and destination snapshots (thus the checksum differs), which means it changes during the incremental sync. I think that's because it's a repo and writes something to those files, as the path changes ? 

As our goal is to test the incremental sync, avoiding the logs (a few KBs) altogether, works very well.

Reply:
As I've checked further, this issue doesn't exist in vstart. It happens in teuthology only.

Reply:
> > This happens mainly during the incremental sync. The md5sum differs consistently for the files `.git/logs/HEAD` and `.git/logs/refs/heads/main` for the source and destination snapshots (thus the checksum differs), which means it changes during the incremental sync. I think that's because it's a repo and writes something to those files, as the path changes ?
> 
> But the checksum are for snapshots, which are immutable, so I don't understand how the checksum would _change_ 🤔

@vshankar 
Look at the above instance, there's a difference in .git/refs/heads/giant too.

I think the solution would be to separate the working directory (.git repo) and the dir_root. So that the dir_root would contain the updated ceph-qa-suite without the `.git` dir, ready to sync.

Reply:
> > This happens mainly during the incremental sync. The md5sum differs consistently for the files `.git/logs/HEAD` and `.git/logs/refs/heads/main` for the source and destination snapshots (thus the checksum differs), which means it changes during the incremental sync. I think that's because it's a repo and writes something to those files, as the path changes ?
> 
> But the checksum are for snapshots, which are immutable, so I don't understand how the checksum would _change_ 🤔

It could be because of this issue https://github.com/ceph/ceph/pull/54633#discussion_r1873219699, but as I remember `test_cephfs_mirror_incremental_sync` fails intermittently even before snapdiff was merged.

Anyway, let me know your decision on this. Can I drop this change?

Reply:
Yeh, we need to fix it there rather than working around it.

Reply:
> I think the solution would be to separate the working directory (.git repo) and the dir_root. So that the dir_root would contain the updated ceph-qa-suite without the `.git` dir, ready to sync.

@joscollin I really don't understand why are we tinkering around with .git stuff. The repo was cloned in cephfs and snapshots were taken. So, the snapshotted directory should be correctly replicated irrespective of what lives under that directory.